scio-spanner/src/main/scala/com/spotify/scio/spanner/SpannerIO.scala

```diff
@@ -0,0 +1,120 @@
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.gcp.spanner.SpannerIO.FailureMode
+import org.apache.beam.sdk.io.gcp.spanner.{SpannerConfig, SpannerIO => BSpannerIO}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+sealed trait SpannerIO[T] extends ScioIO[T] {
+  override final val tapT = EmptyTapOf[T]
+  val config: SpannerConfig
+
+  override def testId: String =
+    s"${getClass.getSimpleName}" +
+      s"(${config.getProjectId}, ${config.getInstanceId}, ${config.getDatabaseId})"
+}
+
+object SpannerRead {
+  object ReadParam {
+    private[spanner] val DefaultWithTransaction = false
+    private[spanner] val DefaultWithBatching = true
+  }
+
+  sealed trait ReadMethod
+  case class FromTable(tableName: String, columns: Seq[String]) extends ReadMethod
+  case class FromQuery(query: String) extends ReadMethod
+
+  final case class ReadParam private (
+    readMethod: ReadMethod,
+    withTransaction: Boolean = ReadParam.DefaultWithTransaction,
+    withBatching: Boolean = ReadParam.DefaultWithBatching
+  )
+}
+
+final case class SpannerRead(config: SpannerConfig) extends SpannerIO[Struct] {
+  import SpannerRead._
+
+  override type ReadP = ReadParam
+  override type WriteP = Nothing
+
+  override protected def read(sc: ScioContext, params: ReadP): SCollection[Struct] = sc.wrap {
+    var transform = BSpannerIO
+      .read()
+      .withSpannerConfig(config)
+      .withBatching(params.withBatching)
+
+    transform = params.readMethod match {
+      case x: FromTable => transform.withTable(x.tableName).withColumns(x.columns.asJava)
+      case y: FromQuery => transform.withQuery(y.query)
+    }
+
+    if (params.withTransaction) {
+      val txn = BSpannerIO.createTransaction().withSpannerConfig(config)
+      transform = transform.withTransaction(sc.applyInternal(txn))
+    }
+
+    sc.applyInternal(transform)
+  }
+
+  override protected def write(data: SCollection[Struct], params: WriteP): Future[Tap[Nothing]] =
+    throw new IllegalStateException("SpannerRead is read-only")
+
+  override def tap(params: ReadP): Tap[Nothing] =
+    throw new NotImplementedError("SpannerRead tap is not implemented")
+}
+
+object SpannerWrite {
+  object WriteParam {
+    private[spanner] val DefaultFailureMode = FailureMode.FAIL_FAST
+    private[spanner] val DefaultBatchSizeBytes = 1024L * 1024L
+  }
+
+  final case class WriteParam private (
+    failureMode: FailureMode = WriteParam.DefaultFailureMode,
+    batchSizeBytes: Long = WriteParam.DefaultBatchSizeBytes
+  )
+}
+
+final case class SpannerWrite(config: SpannerConfig) extends SpannerIO[Mutation] {
+  override type ReadP = Nothing
+  override type WriteP = SpannerWrite.WriteParam
+
+  override protected def write(data: SCollection[Mutation],
+                               params: WriteP): Future[Tap[Nothing]] = {
+    var transform = BSpannerIO.write().withSpannerConfig(config)
```

Comment:
@ClaireMcGinty can you do it like this?

```suggestion
    val transform = BSpannerIO
      .write()
      .withSpannerConfig(config)
      .withBatchSizeBytes(params.batchSizeBytes)
      .withFailureMode(params.failureMode)
```